### ISSUE TITLE
Soulshards not being automatically destroyed

### ISSUE THREAD

**Issue by tnekau** [robert388/Necrosis-classic#31]
I have a 16 slot soul bag with shards set to go there. I have Necrosis set to destroy anything above 16 and when the bag is full.

Upon reaching 16 shards in the bag I get a message from necrosis telling me the soul bag is full and further shards will be destroyed, but they are not destroyed and keep accumulating in the other bags until full.

**Comment by k3ymkr** [comment]
So I found this rather confusing as there are two potentially conflicting options.   "Delete once my bag is full" and "Delete more than X".   Personally, I don't see value in more than X if you have a soul bag.   If you don't have a soul bag, I'd think that would be just a different option (an OR not an if).

I changed the block doing the destorying to this, if it helps:

	-- Destroy extra shards (if enabled) || Si il y a un nombre maximum de fragments à conserver, on enlève les supplémentaires
	if NecrosisConfig.DestroyShard
	then
			soulshardbagcount=0;
			for container = 0, 4, 1 do
				if Local.BagIsSoulPouch[container + 1] then
					soulshardbagcount=GetContainerNumSlots(container)					 
				end
			end
			if (Local.Soulshard.Count>soulshardbagcount) then
				for container = 0, 4, 1 do
					if Local.BagIsSoulPouch[container + 1] then break end
					for slot=1, GetContainerNumSlots(container), 1 do
						local itemLink = GetContainerItemLink(container, slot)
						if (itemLink) then
							local _, itemID = strsplit(":", itemLink)
							itemID = tonumber(itemID)
							if (itemID == 6265) then
								if (math.floor(NecrosisConfig.DestroyCount) < GetItemCount(6265)) then
									PickupContainerItem(container, slot)
									if (CursorHasItem()) then
										DeleteCursorItem()
										Local.Soulshard.Count = GetItemCount(6265)
									else
										self:Msg("No item in hand","USER")
									end
								end
								break
							end
						end
					end
					if soulshardbagcount == Local.Soulshard.Count then break end
				end
			end
	end

**Comment by danslo** [comment]
This only _sometimes_ happens to me, and it's solved by doing a `/reload`.

**Comment by tnekau** [comment]
A reload doesn't usually fix it for me(though it does sometimes). It's definitely related to the soul shard bag as if I unequip that it seems OK most of the time.

I've tried changing the options around - unselecting the "put in specific bag" option and delete when full, and just relying on the "no more than x", as well as the other way around.

I don't think it's related to any other addons as it still fails when Necrosis is the only addon running.

**Comment by tnekau** [comment]
Still not working reliably after latest update

**Comment by danslo** [comment]
Try resetting your wtf configs, it doesn't happen to me anymore on recent
update.

**Comment by tnekau** [comment]
Deleting  WTF has helped, it mostly works however sometimes (usually on on login but sometimes just randomly) I receive the following LUA error and it all stops working:

Message: Interface\AddOns\Necrosis\Necrosis.lua:1961: attempt to concatenate a nil value
Time: Sat Sep 21 00:26:48 2019
Count: 1
Stack: Interface\AddOns\Necrosis\Necrosis.lua:1961: attempt to concatenate a nil value
Interface\AddOns\Necrosis\Necrosis.lua:1961: in function `BagExplore'
Interface\AddOns\Necrosis\Initialize.lua:151: in function `Initialize'
Interface\AddOns\Necrosis\Necrosis.lua:355: in function `OnLoad'
[string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>

Locals: self = <table> {
 BindName = <function> defined @Interface\AddOns\Necrosis\Bindings.lua:42
 RezTimerUpdate = <function> defined @Interface\AddOns\Necrosis\Timers\OtherTimers.lua:45
 Unit = <table> {
 }
 InsertTimerParTable = <function> defined @Interface\AddOns\Necrosis\Timers\Functions.lua:46
 ButtonSetup = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:2034
 Timer_Click = <function> defined @Interface\AddOns\Necrosis\XML\Options-Timers.lua:194
 Localize = <table> {
 }
 SoulshardSwitch = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:1970
 FindSlot = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:1996
 Localization_Speech_Es = <function> defined @Interface\AddOns\Necrosis\Speech\Speech-es.lua:37
 BuffVector_Click = <function> defined @Interface\AddOns\Necrosis\XML\Options-Menus.lua:792
 Localization_Dialog_En = <function> defined @Interface\AddOns\Necrosis\Locales\Dialog-en.lua:41
 SetCompanionPage = <function> defined @Interface\AddOns\Necrosis\XML\Options-Buttons.lua:551
 CreateGroup = <function> defined @Interface\AddOns\Necrosis\Timers\GraphicalTimers.lua:45
 Localization_Dialog_De = <function> defined @Interface\AddOns\Necrosis\Locales\Dialog-de.lua:41
 Msg = <function> defined @Interface\AddOns\Necrosis\Message.lua:42
 AddFrame = <function> defined @Interface\AddOns\Necrosis\Timers\GraphicalTimers.lua:114
 HealthstoneCooldown = "Healthstone Cooldown"
 InCombatAttribute = <function> defined @Interface\AddOns\Necrosis\XML\Attributes.lua:443
 RetraitTimerParGuid = <function> defined @Interface\AddOns\Necrosis\Timers\Functions.lua:313
 OnLoad = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:337
 DemonVector_Init = <function> defined @Interface\AddOns\Necrosis\XML\Options-Menus.lua:810
 ShowAntiFearWarning = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:2357
 MenuAttribute = <function> defined @Interface\AddOns\Necrosis\XML\Attributes.lua:47
 CreateWarlockUI = <function> defined @Interface\AddOns\Necrosis\XML\XML.lua:73
 GetCompanionInfo = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:2889
 Event_Init = <function> defined @Interface\AddOns\Necrosis\XML\Options-Sphere.lua:245
 Speech_It = <function> defined @Interface\AddOns\Necrosis\Message.lua:160
 SetTimersConfig = <function> defined @Interface\AddOns\Necrosis\XML\Options-Timers.lua:45
 Event_Click = <function> defined @Interface\AddOns\Necrosis\XML\Options-Sphere.lua:257
 UnitHasBuff = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:2338
 Spell_Init = <function> defined @Interface\AddOns\Necrosis\XML\Options-Sphere.lua:267
 SetMenusConfig = <function> defined @Interface\AddOns\Necrosis\XML\Options-Menus.lua:45
 SetMessagesConfig = <function> defined @Interface\AddOns\Necrosis\XML\Options-Messages.lua:45
 StoneAttribute = <function> defined @Interface\AddOns\Necrosis\XML\Attributes.lua:290
 Spell_Click = <function> defined @Interface\AddOns\Necrosis\XML\Options-Sphere.lua:278
 UpdateHealth = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:1437
 Translation = <table> {
 }
 RegisterManagement = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:695
 ChangeDemon = <function> defined @Interface\AddOns\Necrosis\Necrosis.lua:711
 Speech = <table> {
 }
 Localization_Speech_Ru = <function> defined @Interface\AddOns\Necrosis\Speech\Speech-ru.lua:38
 CreateWarlockPopup = <function> defined @Interface\AddOns\Necrosis\XML\XML.lua:417
 MsgReplace = <function> defined @Interface\AddOns\Necrosis\Message

**Comment by danslo** [comment]
I get this error too (on login), but a `/reload` fixes this for me. 

Probably duplicate of https://github.com/robert388/Necrosis-classic/issues/42 and https://github.com/robert388/Necrosis-classic/issues/44.

**Comment by tnekau** [comment]
Yep, a /reload will generally fix it, however the inconsistent behavior regarding the deleting of soul shards is still persisting. I can't find any consistent way to make it fail, but generally speaking more often than not it does not work. Sometimes a reload will fix it, but more often than not it won't.

When shard management stops working most of the time it's silent (ie no LUA error, just stops deleting).

**Comment by tnekau** [comment]
LUA error still happening as of todays update. Almost every time on login.
